### PR TITLE
[WIP] - Quarkus 2.x Upgrade

### PR DESCRIPTION
**Pull request by pjgg**
**Enhancements:**

- Upgrade to Java 11
- Upgrade to Quarkus 2.x
- Remove `quarkus-universe-bom` from pom file
- Remove avro-maven-plugin
- `quarkus.http.redirect-to-non-application-root-path` was removed and there is no more non-app endpoints redirections

Code-style
- missing `io.quarkus:quarkus-ide-config:2.0.0.Alpha1` dependency

### CODE REVIEW

**Comment by Sgitario** [comment]
I think we should create a branch "1.13" before merging this PR.

File: microprofile/src/main/java/io/quarkus/ts/openshift/microprofile/HelloClient.java [hunk 1]
```diff
@@ -13,7 +13,8 @@
 
 import static java.util.concurrent.CompletableFuture.completedFuture;
 
-@RegisterRestClient(baseUri = "http://microprofile-test:8080/")
+// TODO https://github.com/quarkusio/quarkus/issues/16844
+@RegisterRestClient(baseUri = "http://localhost:8080/")
```
Comment (rsvoboda) [inline microprofile/src/main/java/io/quarkus/ts/openshift/microprofile/HelloClient.java:17]: Why is this change needed?

File: microprofile/src/test/java/io/quarkus/ts/openshift/microprofile/AbstractMicroProfileTest.java [hunk 2]
```diff
@@ -24,7 +25,7 @@ public void hello() {
                 .atMost(59, TimeUnit.SECONDS)
                 .with()
                 .untilAsserted(() -> {
-                    when()
+                    given().log().uri().when()
```
Comment (rsvoboda) [inline microprofile/src/test/java/io/quarkus/ts/openshift/microprofile/AbstractMicroProfileTest.java:28]: is this needed? or just debugging leftover?

File: microprofile/src/test/java/io/quarkus/ts/openshift/microprofile/MicroProfileOpenShiftIT.java [hunk 3]
```diff
@@ -66,14 +66,14 @@ public void verifyTracesInJaeger() {
                     .body("data[0].spans.find { it.operationName == 'GET' }.tags.collect { \"${it.key}=${it.value}\".toString() }", hasItems(
                             "span.kind=client",
                             "component=jaxrs",
-                            "http.url=http://microprofile-test:8080/hello",
+                            "http.url=http://localhost:8080/hello",
```
Comment (rsvoboda) [inline microprofile/src/test/java/io/quarkus/ts/openshift/microprofile/MicroProfileOpenShiftIT.java:69]: same as above, why is this change needed?

**Review by rsvoboda — CHANGES_REQUESTED** [review]

**Comment by rsvoboda** [comment]
@Sgitario why do we need 1.13 branch? There won't be any product based on 1.13.

**Comment by Sgitario** [comment]
> @Sgitario why do we need 1.13 branch? There won't be any product based on 1.13.

Just to keep a branch where the previous changes before Quarkus 2.0. We can also see the git history.

**Comment by rsvoboda** [comment]
I think git history is enough

File: microprofile/src/main/java/io/quarkus/ts/openshift/microprofile/HelloClient.java (hunk 1 above)
Comment (pjgg) [reply microprofile/src/main/java/io/quarkus/ts/openshift/microprofile/HelloClient.java:17]: There is an issue related to `quarkus.application.name` and rest-client. 
https://github.com/quarkusio/quarkus/issues/16844

**Comment by pjgg** [comment]
> I think git history is enough

I agree, I think that we could remove branch 1.13 and rely on the git history.

File: pom.xml [hunk 4]
```diff
@@ -68,12 +69,13 @@
...
         <version.jandex-maven-plugin>1.0.8</version.jandex-maven-plugin>
         <version.jjwt>0.11.2</version.jjwt>
+        <version.quarkus.qpid.jms>0.24.0</version.quarkus.qpid.jms>
         <version.apache.avro>1.10.2</version.apache.avro>
         <version.keytool-maven-plugin>1.5</version.keytool-maven-plugin>
         <version.maven-compiler-plugin>3.8.1</version.maven-compiler-plugin>
         <version.maven-jar-plugin>3.2.0</version.maven-jar-plugin>
         <version.maven-surefire-plugin>2.22.2</version.maven-surefire-plugin>
-        <version.quarkus>1.13.3.Final</version.quarkus>
+        <version.quarkus>2.0.0.Alpha1</version.quarkus>
```
Comment (rsvoboda) [inline pom.xml:78]: Use Alpha2

**Review by rsvoboda — CHANGES_REQUESTED** [review]
Strimzi/Kafka - `Fixed on upstream` means latest Quarkus main?

Remove avro-maven-plugin ... why?

**Comment by pjgg** [comment]
> Strimzi/Kafka - `Fixed on upstream` means latest Quarkus main?
> 
> Remove avro-maven-plugin ... why?

 - Kafka/strimzi fixed on `2.0.0.Alpha2`
 -  avro-maven-plugin -> is not required anymore, is integrated in `quarkus-maven-plugin`
 
 ```
 <plugin>
      <groupId>io.quarkus</groupId>
      <artifactId>quarkus-maven-plugin</artifactId>
      <version>${version.plugin.quarkus}</version>
      <executions>
          <execution>
                 <goals>
                     <goal>build</goal>
                      <goal>generate-code</goal>
                </goals>
          </execution>
      </executions>
 </plugin>
 ```
You just need to add `generate-code` as a goal, and that is all!.

File: pom.xml [hunk 5]
```diff
@@ -34,7 +34,8 @@
         <module>security/jwt</module>
         <module>security/jwt-cookie</module>
         <module>security/keycloak</module>
-        <module>security/keycloak-authz</module>
+        <!-- TODO: disabled keycloak-authz https://github.com/quarkusio/quarkus/issues/17222 -->
+        <!-- <module>security/keycloak-authz</module> -->
```
Comment (Sgitario) [inline pom.xml:38]: We don't need to exclude the whole module, but just adding the `@Authenticated` annotation to the resources.
Also, this issue is duplicated, the right issue is https://github.com/quarkusio/quarkus/issues/17164

**Comment by pjgg** [comment]
moved to: #312
